lib/config/migration.ts

```diff
@@ -97,11 +97,20 @@ export function migrateConfig(
         migratedConfig.hostType = val;
         delete migratedConfig.platform;
       } else if (parentKey === 'hostRules' && key === 'endpoint') {
-        migratedConfig.baseUrl = val;
+        migratedConfig.matchHost ||= val;
         delete migratedConfig.endpoint;
       } else if (parentKey === 'hostRules' && key === 'host') {
-        migratedConfig.hostName = val;
+        migratedConfig.matchHost ||= val;
         delete migratedConfig.host;
+      } else if (parentKey === 'hostRules' && key === 'baseUrl') {
+        migratedConfig.matchHost ||= val;
+        delete migratedConfig.baseUrl;
+      } else if (parentKey === 'hostRules' && key === 'hostName') {
+        migratedConfig.matchHost ||= val;
+        delete migratedConfig.baseUrl;
```

Comment:
@rarkins shouldn't it be `delete migratedConfig.hostName` ?

Reply:
Yes, but it's already been fixed: https://github.com/renovatebot/renovate/blob/83ab94eba680f6705dbbdbb60207620eeec06a76/lib/config/migration.ts#L113